### PR TITLE
Ensure ascii

### PR DESCRIPTION
**Pull request by MatProv**
Validates user inputs to block non-ascii characters.

"/" and " " characters are removed as well to prevent API call issues.

Validation applied to : group_name, service_name and username.
Other URL parameters are auto-generated (eg: ids) so we don't need to validate them.

Related to issues [#84](https://github.com/Ouranosinc/Magpie/issues/84) and [#208](https://github.com/Ouranosinc/Magpie/issues/208)

### CODE REVIEW

File: magpie/ui/management/views.py [hunk 1]
```diff
@@ -815,11 +834,30 @@ def add_service(self):
         cur_svc_type = self.request.matchdict["cur_svc_type"]
         svc_types, cur_svc_type, services = self.get_services(cur_svc_type)
 
+        services_keys_sorted = self.get_service_types()
+        services_phoenix_indices = [(1 if services_keys_sorted[i] in register.SERVICES_PHOENIX_ALLOWED else 0)
+                                    for i in range(len(services_keys_sorted))]
+
+        return_data = {u"invalid_service_name": False, u"invalid_service_url": False, u"form_service_name": u"", u"form_service_url": u"", u"cur_svc_type": cur_svc_type, u"service_types": svc_types, u"services_phoenix": register.SERVICES_PHOENIX_ALLOWED, u"services_phoenix_indices": services_phoenix_indices}
```
Comment (fmigneault) [inline magpie/ui/management/views.py]: Don't redefine the whole dict, just update keys of one defined at func start.

Comment (MatProv) [reply magpie/ui/management/views.py]: It was defined at the end of the function and moved up. 
It has been moved up to allow return_data to be sent without missing values when error occurs.

File: magpie/api/management/group/group_views.py [hunk 2]
```diff
@@ -26,6 +26,8 @@ def get_groups_view(request):
 def create_group_view(request):
     """Create a group."""
     group_name = ar.get_value_multiformat_post_checked(request, "group_name")
+    ax.verify_param(group_name, notNone=True, notEmpty=True, httpError=HTTPBadRequest,
+                    msgOnFail=s.Group_POST_Name_BadRequestResponseSchema.description, isAlpha=True)
```
Comment (fmigneault) [inline magpie/api/management/group/group_views.py]: put it directly in `create_group` so everywhere inherits it

File: magpie/api/management/service/service_views.py [hunk 3]
```diff
@@ -70,6 +70,7 @@ def register_service_view(request):
     service_url = ar.get_value_multiformat_post_checked(request, "service_url")
     service_type = ar.get_value_multiformat_post_checked(request, "service_type")
     service_push = asbool(ar.get_multiformat_post(request, "service_push"))
+    ax.verify_param(service_name, httpError=HTTPBadRequest, msgOnFail=s.Services_POST_BadRequestResponseSchema.description, isAlpha=True)
```
Comment (fmigneault) [inline magpie/api/management/service/service_views.py]: Will indicate line too long during PR evaluation (other places also)
run `make lint` to validate any static code checks before push

File: magpie/ui/utils.py [hunk 4]
```diff
@@ -2,6 +2,7 @@
 from magpie.utils import get_header, CONTENT_TYPE_JSON
 from typing import TYPE_CHECKING
 import json
+import re
```
Comment (fmigneault) [inline magpie/ui/utils.py]: not needed anymore

File: magpie/ui/utils.py [hunk 5]
```diff
@@ -70,3 +71,7 @@ def wrap(*args, **kwargs):
         except Exception as e:
             raise HTTPBadRequest(detail=str(e))
     return wrap
+
+
+def invalid_url_param(str):
+    return not str.isalpha();
```
Comment (fmigneault) [inline magpie/ui/utils.py]: missing endline

File: magpie/utils.py [hunk 6]
```diff
@@ -340,10 +340,4 @@ def get(cls, key_or_value, default=None):
...
                 return m_val
-        return default
-
-def invalid_url_param(str):
-    return not is_ascii(str) or "/" in str or " " in str
-
-def is_ascii(str): 
-    return re.match('^[\x00-\x7F]+$', str)
\ No newline at end of file
+        return default
```
Comment (fmigneault) [inline magpie/utils.py]: missing endline

**Comment by fmigneault** [comment]
@MatProv needs rebase also it seems (conflicts)

**Comment by fmigneault** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 16
           

Complexity increasing per file
==============================
- magpie/ui/management/views.py  1
- magpie/api/exception.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/fmigneault/Magpie/pullRequest?prid=4650807&bid=15573107)

File: magpie/utils.py [hunk 7]
```diff
@@ -29,15 +32,16 @@
 
 CONTENT_TYPE_ANY = "*/*"
```
Comment (fmigneault) [inline magpie/utils.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/app/fmigneault/Magpie/pullRequest?prid=4650807)

File: magpie/ui/utils.py [hunk 8]
```diff
@@ -21,8 +22,8 @@ def request_api(request,            # type: Request
                 cookies=None,       # type: Optional[CookiesType]
                 ):                  # type: (...) -> Response
     """
-    Use a pyramid sub-request to request Magpie API routes via the UI.
-    This avoids max retries and closed connections when using 1 worker (eg: during tests).
+    Use a pyramid sub-request to request Magpie API routes via the UI. This avoids max retries and closed connections
+    when using 1 worker (eg: during tests).
```
Comment (fmigneault) [inline magpie/ui/utils.py]: Codacy Issue found: [Unnecessary semicolon](https://app.codacy.com/app/fmigneault/Magpie/pullRequest?prid=4650807)

File: magpie/ui/management/views.py [hunk 9]
```diff
@@ -586,18 +596,27 @@ def edit_group(self):
             res_id = self.request.POST.get("resource_id")
             group_path = schemas.GroupAPI.path.format(group_name=group_name)
 
+            group_info["invalid_group_name"] = False
+
             if u"delete" in self.request.POST:
                 resp = request_api(self.request, group_path, "DELETE")
                 check_response(resp)
```
Comment (fmigneault) [inline magpie/ui/management/views.py]: Codacy Issue found: [line too long (309 > 159 characters)](https://app.codacy.com/app/fmigneault/Magpie/pullRequest?prid=4650807)

File: magpie/api/management/service/service_views.py [hunk 10]
```diff
@@ -21,22 +21,28 @@
 @s.ServiceTypesAPI.get(tags=[s.ServicesTag], response_schemas=s.ServiceTypes_GET_responses)
 @view_config(route_name=s.ServiceTypesAPI.name, request_method="GET")
 def get_service_types_view(request):
-    """List all available service types."""
```
Comment (fmigneault) [inline magpie/api/management/service/service_views.py]: Codacy Issue found: [Line too long (137/100)](https://app.codacy.com/app/fmigneault/Magpie/pullRequest?prid=4650807)

File: magpie/api/management/service/service_views.py [hunk 11]
```diff
@@ -21,22 +21,28 @@
...
+    """
+    List all available service types.
+    """
     return ax.valid_http(httpSuccess=HTTPOk, content={u"service_types": list(sorted(SERVICE_TYPE_DICT.keys()))},
                          detail=s.ServiceTypes_GET_OkResponseSchema.description)
 
 
 @s.ServiceTypeAPI.get(tags=[s.ServicesTag], response_schemas=s.ServiceType_GET_responses)
 @view_config(route_name=s.ServiceTypeAPI.name, request_method="GET")
```
Comment (fmigneault) [inline magpie/api/management/service/service_views.py]: Codacy Issue found: [Line too long (111/100)](https://app.codacy.com/app/fmigneault/Magpie/pullRequest?prid=4650807)

File: magpie/ui/management/views.py [hunk 12]
```diff
@@ -815,11 +835,30 @@ def add_service(self):
...
+        return_data = {u"invalid_service_name": False, u"invalid_service_url": False, u"form_service_name": u"", u"form_service_url": u"", u"cur_svc_type": cur_svc_type, u"service_types": svc_types, u"services_phoenix": register.SERVICES_PHOENIX_ALLOWED, u"services_phoenix_indices": services_phoenix_indices}
+        check_data = [u"invalid_service_name", u"invalid_service_url"]
+
         if "register" in self.request.POST:
             service_name = self.request.POST.get("service_name")
             service_url = self.request.POST.get("service_url")
             service_type = self.request.POST.get("service_type")
             service_push = self.request.POST.get("service_push")
+            return_data[u"form_service_name"] = service_name
+            return_data[u"form_service_url"] = service_url
```
Comment (fmigneault) [inline magpie/ui/management/views.py]: Codacy Issue found: [Final newline missing](https://app.codacy.com/app/fmigneault/Magpie/pullRequest?prid=4650807)

File: magpie/api/exception.py [hunk 13]
```diff
@@ -110,6 +111,8 @@ def verify_param(   # noqa: E126
             raise TypeError("'isEqual' is not a 'bool'")
         if not isinstance(ofType, bool):
             raise TypeError("'ofType' is not a 'bool'")
+        if not isinstance(isAlpha, bool):
+            raise TypeError("'isAlpha' is not a 'bool'")
```
Comment (fmigneault) [inline magpie/api/exception.py]: Codacy Issue found: [Line too long (123/100)](https://app.codacy.com/app/fmigneault/Magpie/pullRequest?prid=4650807)

File: magpie/utils.py [hunk 14]
```diff
@@ -29,15 +32,16 @@
...
 CONTENT_TYPE_JSON = "application/json"
```
Comment (fmigneault) [inline magpie/utils.py]: Codacy Issue found: [undefined name 're'](https://app.codacy.com/app/fmigneault/Magpie/pullRequest?prid=4650807)

File: magpie/utils.py (hunk 7 above)
Comment (fmigneault) [inline magpie/utils.py]: Codacy Issue found: [Redefining built-in 'str'](https://app.codacy.com/app/fmigneault/Magpie/pullRequest?prid=4650807)

File: magpie/ui/management/views.py (hunk 9 above)
Comment (fmigneault) [inline magpie/ui/management/views.py]: Codacy Issue found: [Line too long (309/100)](https://app.codacy.com/app/fmigneault/Magpie/pullRequest?prid=4650807)

File: magpie/ui/utils.py (hunk 8 above)
Comment (fmigneault) [inline magpie/ui/utils.py]: Codacy Issue found: [Final newline missing](https://app.codacy.com/app/fmigneault/Magpie/pullRequest?prid=4650807)

File: magpie/ui/utils.py (hunk 4 above)
Comment (fmigneault) [inline magpie/ui/utils.py]: Codacy Issue found: [Unused import re](https://app.codacy.com/app/fmigneault/Magpie/pullRequest?prid=4650807)

Comment (fmigneault) [inline magpie/ui/utils.py]: Codacy Issue found: ['re' imported but unused](https://app.codacy.com/app/fmigneault/Magpie/pullRequest?prid=4650807)

File: magpie/ui/management/views.py [hunk 15]
```diff
@@ -751,7 +771,7 @@ def add_remote_resource(self, service_type, services_names, user_or_group, remot
         for remote_resource in parents:
             name = remote_resource.resource_name
```
Comment (fmigneault) [inline magpie/ui/management/views.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/app/fmigneault/Magpie/pullRequest?prid=4650807)

File: magpie/ui/utils.py [hunk 16]
```diff
@@ -21,8 +22,8 @@ def request_api(request,            # type: Request
                 cookies=None,       # type: Optional[CookiesType]
                 ):                  # type: (...) -> Response
     """
-    Use a pyramid sub-request to request Magpie API routes via the UI.
-    This avoids max retries and closed connections when using 1 worker (eg: during tests).
+    Use a pyramid sub-request to request Magpie API routes via the UI. This avoids max retries and closed connections
```
Comment (fmigneault) [inline magpie/ui/utils.py]: Codacy Issue found: [Redefining built-in 'str'](https://app.codacy.com/app/fmigneault/Magpie/pullRequest?prid=4650807)

File: magpie/utils.py [hunk 17]
```diff
@@ -19,6 +21,7 @@
 import six
 import sys
 import os
+
 if TYPE_CHECKING:
     from magpie.definitions.typedefs import (  # noqa: F401
```
Comment (fmigneault) [inline magpie/utils.py]: Codacy Issue found: [Redefining built-in 'str'](https://app.codacy.com/app/fmigneault/Magpie/pullRequest?prid=4650807)

**Pull request closed without merging**